lib/fizzy/execute.cpp

```diff
@@ -140,14 +140,15 @@ inline uint32_t grow_memory(
 {
     const auto cur_pages = memory.size() / PageSize;
     assert(cur_pages <= size_t(std::numeric_limits<int32_t>::max()));
-    const auto new_pages = cur_pages + delta_pages;
+    const auto new_pages = uint64_t{cur_pages} + delta_pages;
```

Comment:
Why? It would be redundant as operands are promoted to biggest type, it's similar to what we did here:
https://github.com/wasmx/fizzy/blob/0c663e480e0fcc2b91ece9301671ca4593f78cf9/lib/fizzy/execute.cpp#L74